Monitoring

dkistner opened the pull request:
Add basic monitoring capabilities

- Add Prometheus, Alertmanager and Grafana monitoring stack
- Add proper prometheus scrape configs for cluster and controlplane components
- Add basic alert definitions for cluster and controlplane components

**Open issues:**
Added scrape target for the etcds. Etcds "/metrics" endpoints can't be scraped, because it returns a binary output. This is not a blocker, but when we accept this PR, we should open an issue to address this.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/gardener/kubify?pullRequest=8) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/gardener/kubify?pullRequest=8) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: mandelsoft<br/>:x: dkistner<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/gardener/kubify?pullRequest=8) it.</sub>

dkistner commented:
@afritzler, @mandelsoft Could you please review?

afritzler commented:
I already had a look at it in the morning. Looks good to me. However I would put some documentation/example either into [1] or into a separate file inside the docs folder. 

[1] https://github.com/gardener/kubify/blob/master/doc/addons.md

afritzler commented:
Just ran the test now: for me Grafana does not have the Prometheus datasource configured.

dkistner commented:
I added a default prometheus datasource to Grafana. 
Furthermore I added the monitoring stack to list of available addons in the documentation. I think further docu is required.

Good Point: I will add a Grafana dashboard which displays the basic metrics, which I have already defined in Prometheus.

mandelsoft commented:
Unfortunately terraform ALWAYS freshly creates the templates, no -/+ but always a +.
This needs further investigation.

afritzler commented:
Btw. the monitoring addon seems to be mandatory now. If I comment it out of the addons map the thing crashes.

dkistner commented:
Add a dashboard with basic cluster health metrics.
Rebased master branch, which contains a fix for the issue which @mandelsoft mentioned.
The monitoring stack addons are not longer mandatory.
